src/library/scala/collection/mutable/WrappedArray.scala

```diff
@@ -114,13 +114,75 @@ object WrappedArray {
     case x: Array[Unit]    => new ofUnit(x)
   }).asInstanceOf[WrappedArray[T]]
 
-  implicit def canBuildFrom[T](implicit m: ClassTag[T]): CanBuildFrom[WrappedArray[_], T, WrappedArray[T]] =
+  implicit def canBuildFrom[T](implicit m: ClassTag[T]): CanBuildFrom[WrappedArray[_], T, WrappedArray[T]] = {
+    val tag = implicitly[ClassTag[T]]
+    val cls = tag.runtimeClass
+    (if (cls.isPrimitive) {
+      tag.runtimeClass match {
+        case java.lang.Byte.TYPE => cbfByteArray
+        case java.lang.Short.TYPE => cbfShortArray
+        case java.lang.Character.TYPE => cbfCharArray
+        case java.lang.Integer.TYPE => cbfIntArray
+        case java.lang.Long.TYPE => cbfLongArray
+        case java.lang.Float.TYPE => cbfFloatArray
+        case java.lang.Double.TYPE => cbfDoubleArray
+        case java.lang.Boolean.TYPE => cbfBooleanArray
+        case java.lang.Void.TYPE => cbfUnitArray
+      }
+    } else if (cls == ObjectClass) {
+      cbfObjectArray
+    } else {
+      refCBF[T with AnyRef](tag.asInstanceOf[ClassTag[T with AnyRef]])
+    }).asInstanceOf[CanBuildFrom[WrappedArray[_], T, WrappedArray[T]]]
+  }
+
+  private[this] val ObjectClass = classOf[Object]
+
+  private[this] val cbfBooleanArray = new CanBuildFrom[WrappedArray[_], Boolean, WrappedArray[Boolean]] {
+    def apply(from: WrappedArray[_]) = new ArrayBuilder.ofBoolean mapResult WrappedArray.make[Boolean]
+    def apply = new ArrayBuilder.ofBoolean mapResult WrappedArray.make[Boolean]
+  }
```

Comment:
It would be nice to have `CanBuildFrom:mapResult` but I guess it's too late for that now.

Reply:
maybe in 2.14?